packages/orchestration/src/orchestration-client.ts

```diff
@@ -50,17 +53,57 @@ export class OrchestrationClient {
 
     return new OrchestrationResponse(response);
   }
+
+  async stream(
+    prompt?: Prompt,
+    requestConfig?: CustomRequestConfig,
+    controller = new AbortController()
+  ): Promise<OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>> {
+    const response = new OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>();
+    response.stream = (await this.createStream(controller, prompt, requestConfig))
+      ._pipe(OrchestrationChatCompletionStream._processChunk)
+      ._pipe(OrchestrationChatCompletionStream._processFinishReason, response)
+      ._pipe(OrchestrationChatCompletionStream._processTokenUsage, response);
+    return response;
+  }
+
+  private async createStream(
+    controller: AbortController,
+    prompt?: Prompt,
+    requestConfig?: CustomRequestConfig
+  ): Promise<OrchestrationChatCompletionStream<CompletionPostResponseStreaming>> {
+    const body = constructCompletionPostRequest(this.config, prompt, true);
+    const deploymentId = await resolveDeploymentId({
+      scenarioId: 'orchestration',
+      resourceGroup: this.deploymentConfig?.resourceGroup
+    });
+
+    const response = await executeRequest(
+      {
+        url: `/inference/deployments/${deploymentId}/completion`,
+        resourceGroup: this.deploymentConfig?.resourceGroup
+      },
+      body,
+      {
+        ...requestConfig,
+        signal: controller.signal
+      }
+    );
+    return OrchestrationChatCompletionStream._create(response, controller);
+  };
 }
 
 /**
  * @internal
  */
 export function constructCompletionPostRequest(
   config: OrchestrationModuleConfig,
-  prompt?: Prompt
+  prompt?: Prompt,
+  stream = false
 ): CompletionPostRequest {
   return {
     orchestration_config: {
+      stream,
```

Comment:
[q] Do you know if orchestration also needs `stream_options` to include token usage in the last chunk?

```ts
stream_options: {
  include_usage: true
}
```

Reply:
I didn't find anything in their docs.
They do have streaming options for specific modules, but didn't specify any usage related options.

Related docs: https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/streaming?locale=en-US